scripted/sbt/src/main/scala/sbt/test/ScriptedTests.scala

```diff
@@ -8,120 +8,277 @@ package test
+      |    val nameScriptedSetting = name.in(LocalRootProject).:=(
+      |        if (name.value.startsWith("sbt_")) "$testName" else name.value)
+      |    val state1 = Project.extract(state0).append(nameScriptedSetting, state0)
+      |    "initialize" :: state1
+      |  }
+      |}
+    """.stripMargin
+
+  /** Defines the batch execution of scripted tests.
+   *
+   * Scripted tests are run one after the other one recycling the handlers, under
+   * the assumption that handlers do not produce side effects that can change scripted
+   * tests' behaviours.
+   *
+   * In batch mode, the test runner performs these operations between executions:
+   *
+   * 1. Delete previous test files in the common test directory.
+   * 2. Copy over next test files to the common test directory.
+   * 3. Reload the sbt handler.
+   *
+   * @param groupedTests The labels and directories of the tests to run.
+   * @param tempTestDir The common test directory.
+   * @param preHook The hook to run before scripted execution.
+   * @param log The logger.
+   */
+  private def runBatchedTests(
+      groupedTests: Seq[((String, String), File)],
+      tempTestDir: File,
+      preHook: File => Unit,
+      log: Logger
+  ): Seq[Option[String]] = {
+
+    val runner = new BatchScriptRunner
+    val buffer = new BufferedLogger(new FullLogger(log))
+    val handlers = createScriptedHandlers(tempTestDir, buffer)
+    val states = new BatchScriptRunner.States
+    val seqHandlers = handlers.values.toList
+    runner.initStates(states, seqHandlers)
+
+    def runBatchTests = {
+      groupedTests.map {
+        case ((group, name), originalDir) =>
+          val label = s"$group / $name"
+          println(s"Running $label")
+          // Copy test's contents and reload the sbt instance to pick them up
+          IO.copyDirectory(originalDir, tempTestDir)
+
+          val runTest = () => {
+            // Reload and initialize (to reload contents of .sbtrc files)
+            val pluginImplementation = createAutoPlugin(name)
+            IO.write(tempTestDir / "project" / "InstrumentScripted.scala", pluginImplementation)
+            val sbtHandler = handlers.getOrElse('>', sys.error("Missing sbt handler."))
+            val commandsToRun = ";reload;setUpScripted"
+            val statement = Statement(commandsToRun, Nil, successExpected = true, line = -1)
+            // Run reload inside the hook to reuse error handling for pending tests
+            val wrapHook = (file: File) => {
+              preHook(file)
+              runner.processStatement(sbtHandler.asInstanceOf[SbtHandler], statement, states)
+            }
+            commonRunTest(label, tempTestDir, wrapHook, handlers, runner, states, buffer)
+          }
+
+          // Run the test and delete files (except global that holds local scala jars)
+          val result = runOrHandleDisabled(label, tempTestDir, runTest, buffer)
+          IO.delete(tempTestDir.*("*" -- "global").get)
```

Comment:
This wipes out the forked instances temp directory? So you're saving time by not resolving sbt 1.0.0-SNAPSHOT from the second batch onwards?